### PR TITLE
Remove unnecessary Helm Chart deployment

### PR DESCRIPTION
**Pull request by handlerww**
### What problem does this PR solve?
Ref #3652

### Code changes

- [x] Has Go code change
- [x] Has CI related scripts change

### Tests

- [ ] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
None
```

### CODE REVIEW

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 1]
```diff
@@ -310,342 +331,343 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
-		tcName := tcCfg.ClusterName
-
-		oldSvc, err := c.CoreV1().Services(ns).Get(controller.TiDBMemberName(tcName), metav1.GetOptions{})
-		framework.ExpectNoError(err, "failed to get service for TidbCluster: %v", tcCfg)
-		tc, err := cli.PingcapV1alpha1().TidbClusters(ns).Get(tcName, metav1.GetOptions{})
-		framework.ExpectNoError(err, "failed to get TidbCluster: %v", tcCfg)
-		if isNil, err := gomega.BeNil().Match(metav1.GetControllerOf(oldSvc)); !isNil {
-			log.Failf("Expected TiDB service created by helm chart is orphaned: %v", err)
-		}
+	ginkgo.Context("[Feature: Helm Chart migrate to CR]", func() {
```
Comment (handlerww) [inline tests/e2e/tidbcluster/tidbcluster.go:334]: There are 3 cases using tidb-chart. I put these into this context.

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 2]
```diff
@@ -251,27 +251,42 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
 		}
 
 		ginkgo.By("Deploy initial tc")
-		tcCfg := newTidbClusterConfig(e2econfig.TestConfig, ns, "host-network", "", utilimage.TiDBV4)
-		tcCfg.Resources["pd.replicas"] = "1"
-		tcCfg.Resources["tidb.replicas"] = "1"
-		tcCfg.Resources["tikv.replicas"] = "1"
-		oa.DeployTidbClusterOrDie(&tcCfg)
-
+		clusterName := "host-network"
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:254]: What about the other cases in `serial.go` and `stability*.go`?

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 3]
```diff
@@ -251,27 +251,42 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
-		oa.UpgradeTidbClusterOrDie(&tcCfg)
-		oa.CheckTidbClusterStatusOrDie(&tcCfg)
+		// TODO: Considering other components?
+		err = controller.GuaranteedUpdate(genericCli, tc, func() error {
+			tc.Spec.PD.HostNetwork = pointer.BoolPtr(true)
+			tc.Spec.TiKV.HostNetwork = pointer.BoolPtr(true)
+			tc.Spec.TiDB.HostNetwork = pointer.BoolPtr(true)
+			return nil
+		})
+		framework.ExpectNoError(err, "failed to scale out TidbCluster: %q", tc.Name)
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:273]: Please revise the message and also check the other messages.

Comment (handlerww) [reply tests/e2e/tidbcluster/tidbcluster.go:273]: fixed

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 4]
```diff
@@ -881,19 +903,21 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
 	ginkgo.Context("[Feature: AutoFailover]", func() {
 		// TODO: explain purpose of this case
 		ginkgo.It("should clear TiDB failureMembers when scale TiDB to zero", func() {
-			ginkgo.By("Deploy initial tc with bad tidb pre-start script")
-			tcCfg := newTidbClusterConfig(e2econfig.TestConfig, ns, "tidb-scale", "admin", utilimage.TiDBV4)
-			tcCfg.Resources["pd.replicas"] = "3"
-			tcCfg.Resources["tikv.replicas"] = "1"
-			tcCfg.Resources["tidb.replicas"] = "1"
-
-			tcCfg.TiDBPreStartScript = strconv.Quote("exit 1")
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:890]: How does the failover happen now?

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 5]
```diff
@@ -881,19 +903,21 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
-			oa.DeployTidbClusterOrDie(&tcCfg)
+			ginkgo.By("Deploy initial tc")
+			tc := fixture.GetTidbCluster(ns, "tidb-scale", utilimage.TiDBV4)
+			tc.Spec.PD.Replicas = 1
+			tc.Spec.TiKV.Replicas = 1
+			tc.Spec.TiDB.Replicas = 1
+			tc.Spec.TiDB.Config.Set("log.file.max-size", "300")
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:911]: What's this for?

Comment (handlerww) [reply tests/e2e/tidbcluster/tidbcluster.go:911]: Use wrong TiDB config to create failureMembers.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 6]
```diff
@@ -284,18 +296,24 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
-		oa.CheckDisasterToleranceOrDie(&tcCfg)
+		ginkgo.By("Deploying tidb cluster")
+		clusterName := "webhook-upgrade-cluster"
+		tc := fixture.GetTidbCluster(ns, clusterName, utilimage.TiDBV4Prev)
+		tc.Spec.PD.Replicas = 3
+		// Deploy
+		err = genericCli.Create(context.TODO(), tc)
+		framework.ExpectNoError(err, "Expected TiDB cluster created")
+		err = oa.WaitForTidbClusterReady(tc, 6*time.Minute, 5*time.Second)
+		framework.ExpectNoError(err, "Expected TiDB cluster ready")
```
Comment (dragonly) [inline tests/e2e/tidbcluster/tidbcluster.go:307]:
```suggestion
		utiltc.MustCreateTCWithComponentsReady(genericCli, oa, tc, 6*time.Minute, 5*time.Second)
```

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 7]
```diff
@@ -311,341 +329,342 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
-			tc.Spec.TiDB.Service.ExternalTrafficPolicy = &trafficPolicy
-			tc.Spec.TiDB.Service.Annotations = map[string]string{
-				"test": "test",
+			oldSvc, err := c.CoreV1().Services(ns).Get(controller.TiDBMemberName(tcName), metav1.GetOptions{})
+			framework.ExpectNoError(err, "failed to get service for TidbCluster: %v", tcCfg)
+			tc, err := cli.PingcapV1alpha1().TidbClusters(ns).Get(tcName, metav1.GetOptions{})
+			framework.ExpectNoError(err, "failed to get TidbCluster: %v", tcCfg)
+			if isNil, err := gomega.BeNil().Match(metav1.GetControllerOf(oldSvc)); !isNil {
+				log.Failf("Expected TiDB service created by helm chart is orphaned: %v", err)
 			}
```
Comment (dragonly) [inline tests/e2e/tidbcluster/tidbcluster.go:351]:
```suggestion
			framework.ExpectEqual(metav1.GetControllerOf(oldSvc), nil, "Expected TiDB service %s/%s created by helm chart is orphaned", ns, oldSvc.Name)
```

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 8]
```diff
@@ -311,341 +329,342 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
+					return false, nil
+				}
+				if isEqual, err := gomega.Equal(trafficPolicy).Match(svc.Spec.ExternalTrafficPolicy); !isEqual {
+					log.Logf("tidb service traffic policy is not %s, %v", svc.Spec.ExternalTrafficPolicy, err)
+					return false, nil
+				}
+				if haveKV, err := gomega.HaveKeyWithValue("test", "test").Match(svc.Annotations); !haveKV {
+					log.Logf("tidb service has no annotation test=test, %v", err)
+					return false, nil
+				}
```
Comment (dragonly) [inline tests/e2e/tidbcluster/tidbcluster.go:412]: suggest change to plain golang comparison for better readability, like `if corev1.ServiceTypeNodePort != svc.Spec.Type`

**Pull request merged**